### PR TITLE
Feedback

### PR DESCRIPTION
**Pull request by samhstn**
Merge after #77 

Ref #62, #59 

- [x] Alter form inputs so we can have [`richtext`](http://docs.wagtail.io/en/v1.11.1/topics/writing_templates.html#rich-text-filter)
- [x] Pull through before_input and after_input (manually)
- [x] Template matches `app`
- [x] Form data is sent through to wagtail admin (no more spreadsheets)
- [x] Feedback form looks the same as `app`
- [x] Write up why it is difficult to we cannot pull through custom form fields and how we got around it (so no one will spend this time in the future trying to configure this when it's not possible) ([below](https://github.com/LDMW/cms/pull/69#issuecomment-317399292))
- [x] Feedback form has all fields as before
- [x] Feedback form has error handling for invalid and required fields
- [x] Error handling is styled as before
- [x] Feedback form has success message
- [x] Ensure tidy code
- [x] Form inputs remain after submitting invalid request
- [x] Add tests for this module

### CODE REVIEW

**Review by nelsonic — APPROVED** [review]
@shouston3 looks good so far 👍

**Comment by samhstn** [comment]
# Custom form not easy to pull through in Wagtail

After opening a SA question on this [here](https://stackoverflow.com/questions/45212983/wagtail-not-pulling-through-custom-field-panels), I have had no replies, which seems to validate that it is not really practical to do.

After overloading Wagtails [`AbstractFormField`](https://github.com/wagtail/wagtail/blob/master/wagtail/wagtailforms/models.py#L78) as well as the [`AbstractForm`](https://github.com/wagtail/wagtail/blob/master/wagtail/wagtailforms/models.py#L155) [here](https://github.com/wagtail/wagtail/blob/240fa9153b86e641497a521ee135b28dfcf1a15b/wagtail/wagtailforms/forms.py#L111) and [here](https://github.com/LDMW/cms/pull/69/files#diff-531250345e189b774f295d5de5d89e47R161)

To add in our custom form fields [here](https://github.com/wagtail/wagtail/blob/240fa9153b86e641497a521ee135b28dfcf1a15b/wagtail/wagtailforms/forms.py#L111) and [here](https://github.com/LDMW/cms/pull/69/files#diff-531250345e189b774f295d5de5d89e47R161) results in there being an `__init__` error in the [`Field validation`](https://github.com/django/django/blob/5debbdfcc84266703191e084914998e38f5f52eb/django/forms/fields.py#L58)

@reddog and I decided that it was not meant to be changed in this way and instead to just pull it through to the front manually

File: feedback/models.py [hunk 1]
```diff
@@ -1,49 +1,159 @@
...
+from django.shortcuts import render
+
+from modelcluster.fields import ParentalKey
+
+from django.utils.translation import ugettext_lazy as _
+
+from django.utils.translation import ugettext_lazy as _
+from django.shortcuts import render
+
+from django.utils.translation import ugettext_lazy as _
```
Comment (nelsonic) [inline feedback/models.py]: do we _need_ to import `ugettext_lazy` more than once?

File: likes/models.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from django.db import models
+
+class Likes(models.Model):
+    user_hash = models.CharField(max_length=255)
+    resource = models.ForeignKey(
+        'resources.ResourcePage',
+        on_delete=models.CASCADE, # When resource is deleted, delete the likes as well
```
Comment (nelsonic) [inline likes/models.py:7]: 👍

**Review by nelsonic — APPROVED** [review]
@shouston3 & @Danwhy great work! happy to merge this (_only had one question, please check then re-assign..._) 👍

**Review by nelsonic — COMMENTED** [review]
@shouston3 looks great! thanks!
Tests: https://github.com/LDMW/cms/pull/69/files#diff-f8794cd9bccf448bc8696d78a0e922c8 could use some additional cases but for now it's fine. 👍

**Pull request merged**